### PR TITLE
Update: Implementing Best Practices and Code Refinement

### PR DESCRIPTION
**Pull request by Nisarsaeed**

### CODE REVIEW

**Comment by syed-yawar** [comment]
Please write appropriate title for PR which somewhat explains the gist.

File: script.js
Comment (syed-yawar) [inline script.js:1]: please use [IIF](https://www.geeksforgeeks.org/immediately-invoked-function-expressions-iife-in-javascript/) function

File: index.html [hunk 1]
```diff
@@ -20,33 +20,33 @@
...
         <div class="cal-body">
             <div class="row">
-                <button  class="key operator2" onclick="reset()">AC</button>
-                <button  class="key operator2" onclick=" toggleNegate()" >+/-</button>
-                <button  class="key operator2 " onclick="inputOperator('%')">%</button>
-                <button  class="key operator " onclick="inputOperator('÷')">÷</button>
-            </div>
+                <button  class="key operator2" >AC</button>
+                <button  class="key operator2"  >+/-</button>
+                <button  class="key operator2" >%</button>
```
Comment (syed-yawar) [inline index.html]: operator 2 is not very meaningful name
If we want to  only apply styling from this class name then we can add some other meaningful name i.e. yellow-box, light-grey-box, dark-grey-box etc

File: script.js
Comment (syed-yawar) [reply script.js:1]: you can wrap your whole code in 

document.addEventListener("DOMContentLoaded", function() {
(()=>{
//here's your complete code

})()
})

File: script.js [hunk 2]
```diff
@@ -1,66 +1,81 @@
...
+    else if(buttonValue === 'AC'){
+      resetValues();
+    }
+    else if(buttonValue ==='+/-'){
+      toggleNegate();
+    }
+    else{
+      numericKeyClicked(buttonValue);
+    }
+  }
```
Comment (syed-yawar) [inline script.js]:
```suggestion
   const buttonValue = event.target.textContent;

  switch (buttonValue) {
    case '+':
    case '−':
    case '×':
    case '÷':
    case '%':
      operatorClicked(buttonValue);
      break;
    case '=':
      calculate();
      break;
    case 'AC':
      resetValues();
      break;
    case '+/-':
      toggleNegate();
      break;
    default:
      if (/^[0-9]$/.test(buttonValue)) {
        numericKeyClicked(buttonValue);
      }
  }
```

File: script.js [hunk 3]
```diff
@@ -1,66 +1,81 @@
...
+} );
```
Comment (syed-yawar) [inline script.js]: make sure 
- you add new line space after every function
- add new line before return statements if there is other logic in function

File: script.js [hunk 4]
```diff
@@ -1,66 +1,81 @@
...
-    
+    result = parseFloat(currentValue);
+    currentValue = '';
+    currentOperator = operator;   
     for (let i = 0; i < operatorButtons.length; i++) {
       if (operatorButtons[i].textContent === currentOperator ) {
-        operatorButtons[i].style.backgroundColor = "rgb(193 112 0)";
+        operatorButtons[i].classList.add('active-operator');
       }
     }
```
Comment (syed-yawar) [inline script.js]: we can do something like this

```
const activeOperatorButton = Array.from(operatorButtons).find(button => button.textContent === currentOperator);
    if (activeOperatorButton) {
      activeOperatorButton.classList.add('active-operator');
    }
  }
```

File: script.js [hunk 5]
```diff
@@ -1,66 +1,81 @@
...
 let currentOperator = '';
 let result = 0;
-
-function inputNum(value) {
-  currentValue += value;
-  document.getElementById('output').textContent = currentValue;
-}
-
+let displayScreen = document.getElementById('output');
 let operatorButtons = document.getElementsByClassName('operator');
```
Comment (syed-yawar) [inline script.js]: why are we using let here instead of const? are we changing these values somewhere?

File: script.js [hunk 6]
```diff
@@ -1,66 +1,81 @@
...
   currentValue = '';
   currentOperator = '';
   result = 0;
-  Array.from(operatorButtons).forEach(button => {
-    button.style.backgroundColor = '';
-});
-
-  document.getElementById('output').textContent = result;
+  for (let i = 0; i < operatorButtons.length; i++) { 
+      operatorButtons[i].classList.remove('active-operator');
```
Comment (syed-yawar) [inline script.js]: we can use operatorButtons.forEach here, why we removed that?

**Review by syed-yawar — CHANGES_REQUESTED** [review]
Great work Nisar, please go through these comments and update accordingly, Kudos 👏

File: script.js (hunk 2 above)
Comment (Nisarsaeed) [reply script.js]: I have made this change but I can't understand what's the betterment in it as there is no repeation and both switch and if statements are serving exactly the same purpose.

**Review by syed-yawar — APPROVED** [review]
LGTM, Great work. Kudos 👏

**Pull request merged**